applications/product/groovyScripts/facility/facility/CountFacilityInventoryByProduct.groovy

```diff
@@ -55,166 +57,167 @@ rows = [] as ArrayList
 if (action) {
     // ------------------------------
     prodView = new DynamicViewEntity()
-    atpDiffComplexAlias = new ComplexAlias("-")
+    atpDiffComplexAlias = new ComplexAlias('-')
 
-    conditionMap = [facilityId : facilityId]
+    conditionMap = [facilityId: facilityId]
 
     if (offsetQOHQty) {
         try {
             offsetQOH = Integer.parseInt(offsetQOHQty)
             hasOffsetQOH = true
-            searchParameterString = searchParameterString + "&offsetQOHQty=" + offsetQOH
+            searchParameterString = searchParameterString + '&offsetQOHQty=' + offsetQOH
         } catch (NumberFormatException nfe) {
-            logError(nfe, "Caught an exception : " + nfe.toString())
-            request.setAttribute("_ERROR_MESSAGE", "An entered value seems non-numeric")
+            logError(nfe, 'Caught an exception : ' + nfe)
+            request.setAttribute('_ERROR_MESSAGE', 'An entered value seems non-numeric')
         }
     }
     if (offsetATPQty) {
         try {
             offsetATP = Integer.parseInt(offsetATPQty)
             hasOffsetATP = true
-            searchParameterString = searchParameterString + "&offsetATPQty=" + offsetATP
+            searchParameterString = searchParameterString + '&offsetATPQty=' + offsetATP
         } catch (NumberFormatException nfe) {
-            logError(nfe, "Caught an exception : " + nfe.toString())
-            request.setAttribute("_ERROR_MESSAGE", "An entered value seems non-numeric")
+            logError(nfe, 'Caught an exception : ' + nfe)
+            request.setAttribute('_ERROR_MESSAGE', 'An entered value seems non-numeric')
         }
     }
 
-    prodView.addMemberEntity("PRFA", "ProductFacility")
-    prodView.addAlias("PRFA", "productId", null, null, null, Boolean.TRUE, null)
-    prodView.addAlias("PRFA", "minimumStock", null, null, null, Boolean.TRUE, null)
-    prodView.addAlias("PRFA", "reorderQuantity", null, null, null, Boolean.TRUE, null)
-    prodView.addAlias("PRFA", "daysToShip", null, null, null, Boolean.TRUE, null)
-    prodView.addAlias("PRFA", "facilityId", null, null, null, Boolean.TRUE, null)
-
-    prodView.addMemberEntity("PROD", "Product")
-    prodView.addViewLink("PROD", "PRFA", Boolean.FALSE, ModelKeyMap.makeKeyMapList("productId"))
-    prodView.addAlias("PROD", "internalName", null, null, null, Boolean.TRUE, null)
-    prodView.addAlias("PROD", "isVirtual", null, null, null, Boolean.TRUE, null)
-    prodView.addAlias("PROD", "salesDiscontinuationDate", null, null, null, Boolean.TRUE, null)
+/* codenarc-disable */
```

Comment:
If we need to disable codenarc we should be explicit about which rule(s) is being disabled. It will help future authors understand what they could improve.

Reply:
Done with
 https://github.com/gilPts/ofbiz-framework/commit/237c60519505d8b3610a5feb5d33eafa519e4b09